aws-deploy/tasks/main.yml

```diff
@@ -1,25 +1,28 @@
-- name: create launch configuration
-  ec2_lc:
-    region: "{{ region }}"
-    name: "{{ launch_config_name }}"
-    image_id: "{{ image_id }}"
-    key_name: "{{ keypair }}"
-    instance_type: "{{ instance_type }}"
-    security_groups: "{{ security_groups }}"
-    instance_monitoring: yes
-    instance_profile_name: "{{ instance_profile_name }}"
-
-- name: Configure Auto Scaling Group and perform rolling deploy
-  ec2_asg:
-    region: "{{ region }}"
-    name: "{{ auto_scaling_group_name }}"
-    launch_config_name: "{{ launch_config_name }}"
-    health_check_type: ELB
-    health_check_period: 300
-    desired_capacity: "{{ desired_capacity | default(2) }}"
-    min_size: "{{ min_size | default(2) }}"
-    max_size: "{{ max_size | default(2) }}"
-    replace_all_instances: yes
-    replace_batch_size: "{{ replace_batch_size | default(1) }}"
-    state: present
-  register: asg_result
+- name: create launch configuration
+  ec2_lc:
+    region: "{{ region }}"
+    name: "{{ launch_config_name }}"
+    image_id: "{{ image_id }}"
+    key_name: "{{ keypair }}"
+    instance_type: "{{ instance_type }}"
+    security_groups: "{{ security_groups }}"
+    instance_monitoring: yes
+    instance_profile_name: "{{ instance_profile_name }}"
+
+- name: Configure Auto Scaling Group and perform rolling deploy
+  ec2_asg:
+    desired_capacity: "{{ desired_capacity | default(2) }}"
+    health_check_period: 300
+    health_check_type: ELB
+    launch_config_name: "{{ launch_config_name }}"
+    load_balancers: "{{ app_loadbalancers }}"
+    max_size: "{{ max_size | default(2) }}"
+    min_size: "{{ min_size | default(2) }}"
+    name: "{{ auto_scaling_group_name }}"
+    region: "{{ region }}"
+    replace_all_instances: yes
+    replace_batch_size: "{{ replace_batch_size | default(1) }}"
+    state: present
+    tags: "{{ app_tags | default(null) }}"
```

Comment:
:+1: Awesome. And yes, I do agree with you that ideally everything should be in scripts in declarative format but there are certain infrastructure components I'd prefer to avoid touching on every deploy, and ELB is one of them.

I was going to take a stab at adopting your approach but this will help in the interim.